pkg/manager/internal/controllers/project_controller.go

```diff
@@ -0,0 +1,152 @@
+	"github.com/kubermatic/utils/pkg/owner"
+)
+
+const projectNamespaceNameSeparator = "-bulward-"
+
+// ProjectReconciler reconciles a Project object
+type ProjectReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=storage.bulward.io,resources=projects,verbs=get;list;watch;update
+// +kubebuilder:rbac:groups=storage.bulward.io,resources=projects/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=rolebindings,verbs=get;list;watch
+
+func (r *ProjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("Project", req.NamespacedName)
+
+	project := &storagev1alpha1.Project{}
+	if err := r.Get(ctx, req.NamespacedName, project); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if err := r.reconcileNamespace(ctx, log, project); err != nil {
+		return ctrl.Result{}, fmt.Errorf("reconciling namespace: %w", err)
+	}
+
+	if err := r.reconcileMembers(ctx, project); err != nil {
+		return ctrl.Result{}, fmt.Errorf("reconciling members: %w", err)
+	}
+
+	if !project.IsReady() {
+		project.Status.ObservedGeneration = project.Generation
+		project.Status.SetCondition(storagev1alpha1.ProjectCondition{
+			Type:    storagev1alpha1.ProjectReady,
+			Status:  storagev1alpha1.ConditionTrue,
+			Reason:  "SetupComplete",
+			Message: "Project setup is complete.",
+		})
+		if err := r.Status().Update(ctx, project); err != nil {
+			return ctrl.Result{}, fmt.Errorf("updating Projects status: %w", err)
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *ProjectReconciler) reconcileNamespace(ctx context.Context, log logr.Logger, project *storagev1alpha1.Project) error {
+	ns := &corev1.Namespace{}
+	ns.Name = generateProjectNamespaceName(project.Namespace, project.Name)
+
+	if _, err := owner.ReconcileOwnedObjects(ctx, r.Client, log, r.Scheme, project, []runtime.Object{ns}, &corev1.Namespace{}, nil); err != nil {
+		return fmt.Errorf("cannot reconcile namespace: %w", err)
+	}
+
+	if project.Status.Namespace == nil {
+		project.Status.Namespace = &storagev1alpha1.ObjectReference{
+			Name: ns.Name,
+		}
+		if err := r.Status().Update(ctx, project); err != nil {
+			return fmt.Errorf("updating NamespaceName: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *ProjectReconciler) reconcileMembers(ctx context.Context, project *storagev1alpha1.Project) error {
+	rbs := &rbacv1.RoleBindingList{}
+	if err := r.List(ctx, rbs, client.InNamespace(project.Status.Namespace.Name)); err != nil {
+		return fmt.Errorf("list rolebindings: %w", err)
+	}
+	project.Status.Members = extractSubjects(rbs)
+	if err := r.Status().Update(ctx, project); err != nil {
+		return fmt.Errorf("updating members: %w", err)
+	}
+	return nil
+}
+
+// Project Namespace name is organization_name-separator-project_name
+func generateProjectNamespaceName(organizatioName, projectName string) string {
```

Comment:
@nmiculinic @jiachengxu 
Let me know if you have a better idea of a separator(or to do it differently) which will allow us to easily deconstruct the name into Org_namespace + Project_name which is needed for the rolebinding watch. 

For now I used some generic '-bulward-' which i hope wont be something that users will use in org/project names. Eventually we could have some validation on the names to make sure.

Reply:
I've left the comments in other places. Parsing things from strings is tricky, and I'd much prefer avoiding it.